src/components/Typography/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC, useEffect, useState } from 'react'
+
+import { IProps } from './core/typography.interfaces'
+import s from './core/typography.module.scss'
+
+export const Typography: FC<IProps> = ({
+  children,
+  className,
+  font,
+  size,
+  element = 'p',
+  weight,
+  color,
+}) => {
+  const [classNames, setClassNames] = useState<string[]>([])
+
+  useEffect(() => {
+    const classNames = []
+
+    if (font) {
+      classNames.push(s[`typography--font-${font}`])
+    }
+    if (weight) {
+      classNames.push(s[`typography--weight-${weight}`])
+    }
+    if (size) {
+      classNames.push(s[`typography--size-${size}`])
+    }
+    if (color) {
+      classNames.push(s[`typography--color-${color}`])
+    }
+    if (className) {
+      classNames.push(className)
+    }
+
+    setClassNames(classNames)
+  }, [element, font, size, weight, color, className])
```

Comment:
I don't know why I used a state here ^^ Since the computations on classNames are pretty simple I also think the component should remain pure. Using memoization here should even [slow down rendering](https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/).

Reply:
Finally an article that actually answers _when_ to use memoization, thanks!